content/articles/build-ann-with-keras/index.md

```diff
@@ -0,0 +1,247 @@
+```
+
+Now that we have created our model, let use the code below to visualize it:
+
+```python
+from tensorflow.keras.utils import plot_model
+plot_model(ann,
+           to_file="model.png",
+           show_shapes=True,
+           show_layer_names=True,
+          )
+```
+![Artificial Neural Network Plot](/engineering-education/build-ann-with-keras/plot-model.PNG)
+
+We can also use the [NN-SVG](https://alexlenail.me/NN-SVG/) tool to visualize our model.
+
+![ANN Visualization using NN-SVG](/engineering-education/build-ann-with-keras/NN-SVG-architecture.PNG)
+
+### Training the ANN
+
+In training the ANN, we perform a couple of tasks:
+
+- We compile the model with the Adam optimizer.
+- We use the binary cross-entropy loss. 
+- We train the model for 100 epochs. Note that increasing the number of epochs will help your model train better.
+
+```python
+ann.compile(optimizer = 'adam', loss = 'binary_crossentropy', metrics = ['accuracy'])
+ann.fit(X_train, y_train, batch_size = 32, epochs = 100)
+```
+
+![Last 11 epoch of training](/engineering-education/build-ann-with-keras/last-eleven-epochs-of-training.PNG)
+
+### Evaluate the Model
+
+Now that our model training is complete, we can make predictions on a single customer. Let us find out if a customer with the details below will churn.
+
+| Record      | Details     |
+| :---        |    :----:   |
+| Country   | Spain       |
+| Credit Score| 600         |
+| Gender      | Male        |
+| Age         | 40 years  |
+| Tenure      | 3 years     |
+| Balance remaining  | $60000      |
+| Number of Products owned | 2    |
+| Own a Credit Card? | Yes          |
+| Is an Active Member? | Yes        |
+| Estimated Salary | $50000 |
+
+```python
+print(ann.predict(sc.transform([[0, 0, 1, 600, 1, 40, 3, 60000, 2, 1, 1, 50000]])))
+```
+![Predicted Probability of New Customer](/engineering-education/build-ann-with-keras/predicted-probability.PNG)
+
+Remember that after one-hot encoding, the geographical location, `Spain` is represented as `0, 0, 1`. It will be in the first three columns of our matrix of features.
+
+We can add a threshold of 0.5. The customer will leave the bank if the predicted probability is above 0.5. In extreme situations, we can increase the threshold. That is if we want our model to predict `True` only if it is very confident.
+
+```python
+print(ann.predict(sc.transform([[0, 0, 1, 600, 1, 40, 3, 60000, 2, 1, 1, 50000]])) > 0.5)
+```
+![Prediction for New Customer](/engineering-education/build-ann-with-keras/prediction-with-threshold.PNG)
+
+This is great news for the bank! This customer will not churn. Let us assess our model using the test set.
+
+```python
+y_pred = ann.predict(X_test)
+y_pred = (y_pred > 0.5)
+pd.DataFrame(list(zip(y_test, y_pred)), columns=['Actual', 'Predicted'])
+```
+![Actual Values vs Predicted Values](/engineering-education/build-ann-with-keras/actual-vs-predicted-values.PNG)
+
+**Comment on these results obtained**
+
```

Comment:
I've left a comment here in bold. Replace it with the necessary instruction.

Reply:
Done!